File: jmvclibrary/src/main/java/com/gallops/mobile/jmvclibrary/utils/Logger.java
package com.gallops.mobile.jmvclibrary.utils;

import com.gallops.mobile.jmvclibrary.utils.kt.LoggerKt;

/**
 * 日志打印
 * Created by wangyu on 2018/3/27.
 */

public class Logger {
    public static void i(String tag, String msg) {
        LoggerKt.lgI(tag, msg);
    }
    public static void e(String tag, String msg) {
        LoggerKt.lgE(tag, msg);
    }
    public static void d(String tag, String msg) {
        LoggerKt.lgD(tag, msg);
    }
    public static void w(String tag, String msg) {
        LoggerKt.lgW(tag, msg);
    }
}

File: jmvclibrary/src/main/java/com/gallops/mobile/jmvclibrary/utils/kt/Logger.kt
package com.gallops.mobile.jmvclibrary.utils.kt

import android.util.Log
import com.gallops.mobile.jmvclibrary.app.JApp

/**
 * 日志打印
 * Created by wangyu on 2017/7/25.
 */
private inline fun doLog(log: () -> Unit) {
    if (JApp.isDebug()) log()
}

fun lgE(tag: String, msg: String) = doLog { Log.e(tag, msg) }

fun lgI(tag: String, msg: String) = doLog { Log.i(tag, msg) }

fun lgD(tag: String, msg: String) = doLog { Log.d(tag, msg) }

fun lgW(tag: String, msg: String) = doLog { Log.w(tag, msg) }
